sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -140,7 +146,9 @@ case class CreateDataSourceTableAsSelectCommand(
 
     var createMetastoreTable = false
     var existingSchema = Option.empty[StructType]
-    if (sparkSession.sessionState.catalog.tableExists(table.identifier)) {
+    // Pass a table identifier with database part, so that `tableExists` won't check temp views
+    // unexpectedly.
+    if (sparkSession.sessionState.catalog.tableExists(tableIdentWithDB)) {
```

Comment:
I have, in my previous refactor PR: https://github.com/apache/spark/pull/14962

This PR aims to just fix these bugs surgically, so that it's easy to backport to 2.0